modules/utils/graph/Redif.hpp

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <vector>
+
+#include "3rdparty/blaze/Blaze.h"
+#include "modules/distance.hpp"
+
+namespace metric
+{
+	template<typename recType, class Metric = metric::Euclidian<recType>, typename Tv = double>
+	class Redif
+	{
+	public:
+		Redif(std::vector<std::vector<Tv>> trainData, size_t nNeighbors = 10, size_t nIter = 15);

```

Comment:
wrong interface. separate the type factory function with metric and structural parameters as input from pseudo constructor with data set as input